app/main.py

```diff
@@ -1 +1,23 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import (
+    NotVaccinatedError,
+    OutdatedVaccineError,
+    NotWearingMaskError
+)
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
+    masks_needed = 0
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except NotVaccinatedError as e:
+            return str(e)
+        except OutdatedVaccineError as e:
+            return str(e)
```

Comment:
There is no need to create two conditions for exceptions, you could catch only `VaccineError` and since both `NotVaccinatedError` and `OutdatedVaccineError` are inherited from it they will be cached and you'll avoid a duplication.